retriever/endpointfinder/endpointfinder.go

```diff
@@ -30,22 +35,27 @@ type EndpointFinder struct {
 	minerInfoFetcher MinerInfoFetcher
 	h                host.Host
 	httpEndpoints    *lru.Cache[string, []peer.AddrInfo]
+	endpointErrors   *expirable.LRU[string, error]
 }
 
 // NewEndpointFinder returns a new instance of an EndpointFinder
-func NewEndpointFinder(minerInfoFetcher MinerInfoFetcher, h host.Host, size int) (*EndpointFinder, error) {
-	httpEndpoints, err := lru.New[string, []peer.AddrInfo](size)
+func NewEndpointFinder(minerInfoFetcher MinerInfoFetcher, h host.Host, opts ...Option) (*EndpointFinder, error) {
+	cfg := applyOptions(opts...)
+
+	httpEndpoints, err := lru.New[string, []peer.AddrInfo](cfg.LruSize)
```

Comment:
Do we need this to expire also, in case an endpoint changes?

Reply:
yeah, we do, I considered adding it here but it's yet another configuration option and I thought I'd leave the scope a bit more minimal .. but I suppose as long as we're messing around in here.

Reply:
actually, the fun thing about making endpoints also expire is that we get rid of the `error` case so this constructor is even simpler, pushed a version that has that and I put the default expiry at 2 hours, so if an SP changes something on their end we'll eventually catch up with it

Reply:
PTAL at the latest version